flags.go

```diff
@@ -127,26 +161,110 @@ func Enabled(ctx context.Context, name string) (enabled bool) {
 
 	flagsMtx.RLock()
 	defer flagsMtx.RUnlock()
+	enabled = false
 	if flags == nil {
-		enabled = false
+		err = errors.New("empty flags map")
 		return
 	}
 	flag := flags[name]
-
-	// equality should be strict
-	// because Float64() can return 0
-	if f := rand.Float64(); f < flag.Rate {
-		enabled = true
+	if !flag.Active {
 		return
 	}
-	enabled = false
+	for _, r := range flag.Rules {
+
+		res := r.Handle(ctx, properties)
+		var matchBehavior string
+		if res {
+			matchBehavior = r.onMatch()
+		} else {
+			matchBehavior = r.onMiss()
+		}
+		switch matchBehavior {
+		case "on":
+			enabled = true
+			return
+		case "off":
+			enabled = false
+			return
+		case "match_list":
+			continue
+		default:
+			err = errors.New("unknown match behavior: " + matchBehavior)
+			return
+		}
+	}
+	enabled = true
```

Comment:
If we get to the bottom, it should probably be disabled!

Reply:
Fair enough, changed!